amazon-connect/callstats-amazon-connect-shim.js

```diff
@@ -1,6 +1,92 @@
 /*! callstats Amazon SHIM version = 1.1.8 */
 
 (function (global) {
+  class VoiceActivityDetection {
+    constructor(stream, callback) {  
+      this.AudioContext = window.AudioContext || window.webkitAudioContext;
+      this.audioContext = new this.AudioContext();
+      this.analyser = this.audioContext.createAnalyser();
+      this.analyser.fftSize = 512;
+      this.analyser.smoothingTimeConstant = 0.1;
+      this.fftBins = new Float32Array(this.analyser.frequencyBinCount);
+      this.mediaStreamSource = this.audioContext.createMediaStreamSource(stream);
+      this.mediaStreamSource.connect(this.analyser);
+      this.isSpeaking = false;
+      this.maxVolumeHistory = [];
+      for (var i=0; i < 10; i++) {
+        this.maxVolumeHistory.push(0);
+      }
+      this.threshold = -50;
+      this.interval = 50;
+      this.callback = callback;
+      this.start();
+    }
+
+    getMaxVolume () {
+      var maxVolume = -Infinity;
+      this.analyser.getFloatFrequencyData(this.fftBins);
+    
+      for(var i=4, ii=this.fftBins.length; i < ii; i++) {
+        if (this.fftBins[i] > maxVolume && this.fftBins[i] < 0) {
```

Comment:
I don't understand where does this.fftBins[i] < 0 come from? Why does it have to be negative?

Reply:
getFloatFrequencyData(array) gives us the fftbins which are of the range minDecibels - maxDecibels which is -100 to -30